core/store/src/trie/mod.rs

```diff
@@ -411,6 +411,17 @@ pub struct Trie {
     pub flat_state: Option<FlatState>,
 }
 
+/// Trait for reading data from a trie.
+pub trait TrieReader {
+    /// Retrieves node with given key from the trie.
+    ///
+    /// In principle, this does not allow reads data from different chunks (be
+    /// it from different shards or different blocks).  In practice it might be
+    /// possible to retrieve nodes which don’t belong to a trie this reader is
+    /// meant for, but this must not be relied upon.
+    fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, StorageError>;
```

Comment:
It's likely because the use of the trait is quite narrow at the moment.  (My real motivation here was to change check_balance to take just one TrieUpdate object.  I was experimenting with that in pursuit of Trie being a direct field of TrieUpdate rather than using Rc which would simplify some other store-related code).  There may be other places where we create dummy TrieUpdate and those might require addition of other methods as well.